application/server/server.go

```diff
@@ -380,19 +394,24 @@ func TextDocumentDidSaveHandler() jrpc2.Handler {
 		// The context provided by the JSON-RPC server is cancelled once a new message is being processed,
 		// so we don't want to propagate it to functions that start background operations
 		bgCtx := context.Background()
+		logger := log.With().Str("method", "TextDocumentDidSaveHandler").Logger()
 
-		method := "TextDocumentDidSaveHandler"
-		log.Info().Str("method", method).Interface("params", params).Msg("RECEIVING")
+		logger.Info().Interface("params", params).Msg("Receiving")
 		filePath := uri.PathFromUri(params.TextDocument.URI)
 
 		// todo can we push cache management down?
 		f := workspace.Get().GetFolderContaining(filePath)
-		if f != nil {
+		autoScanEnabled := config.CurrentConfig().IsAutoScanEnabled()
+		if f != nil && autoScanEnabled {
 			f.ClearDiagnosticsFromFile(filePath)
```

Comment:
Question: should we clear diagnostics on save even if not auto-scanning? Jetbrains plugins invalidate the results/caches after a save.

Reply:
I think it would be very annoying if deleting a comment in some code file would immediately clear all the diagnostics for the workspace.

Notice that usually there's only one `Folder` for the entire workspace

Reply:
In Eclipse, there is a lot of folders usually in the workspace. But I agree that this has been causing issues every once in a while and I much prefer not to invalidate results.